fish-rust/src/output.rs

```diff
@@ -8,12 +24,706 @@
     }
 }
 
-pub fn output_set_color_support(value: ColorSupport) {
-    extern "C" {
-        pub fn output_set_color_support(value: libc::c_int);
+/// Whether term256 and term24bit are supported.
+static COLOR_SUPPORT: AtomicU8 = AtomicU8::new(0);
+
+/// FFI bits.
+#[no_mangle]
+extern "C" fn output_get_color_support() -> u8 {
+    COLOR_SUPPORT.load(Ordering::Relaxed)
+}
+
+#[no_mangle]
+extern "C" fn output_set_color_support(val: u8) {
+    COLOR_SUPPORT.store(val, Ordering::Relaxed);
+}
+
+/// Returns true if we think tparm can handle outputting a color index.
+fn term_supports_color_natively(c: i32, term: &Term) -> bool {
+    #[allow(clippy::int_plus_one)]
+    if let Some(max_colors) = term.max_colors {
+        max_colors >= c + 1
+    } else {
+        false
+    }
+}
+
+pub fn get_color_support() -> ColorSupport {
+    let val = COLOR_SUPPORT.load(Ordering::Relaxed);
+    ColorSupport::from_bits_truncate(val)
+}
+
+pub fn set_color_support(val: ColorSupport) {
+    COLOR_SUPPORT.store(val.bits(), Ordering::Relaxed);
+}
+
+// These are historic. writembs() attempts to write a multibyte string of type &Option<CString> to the given outputter.
+// writembs() will noisily fail, writembs_nofail will not.
+macro_rules! writembs(
```

Comment:
Ok. This means we'll have a second module with `macro_use` which is fine as long as we don't add a dependency cycle

Reply:
fixed

Reply:
Good point. I'll probably kill off this macro in short order - but I'm glad to have it in at least one commit for bisecting purposes.